tools/StaticAnalysis/Exceptions/AzureRM.Compute/BreakingChangeIssues.csv

```diff
@@ -21,3 +21,75 @@
 "Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.Automation.UpdateAzureRmVmssVM","Update-AzureRmVmssVM","0","3000","The type of property 'CreateOption' of type 'Microsoft.Azure.Commands.Compute.Models.PSVirtualMachineDataDisk' has changed from 'System.String' to 'Microsoft.Azure.Management.Compute.Models.DiskCreateOptionTypes'.","Change the type of property 'CreateOption' back to 'System.String'."
 "Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.Automation.NewAzureRmVmssIpConfigCommand","New-AzureRmVmssIpConfig","0","3000","The type of property 'PrivateIPAddressVersion' of type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSetIPConfiguration' has changed from 'System.String' to 'System.Nullable`1[Microsoft.Azure.Management.Compute.Models.IPVersion]'.","Change the type of property 'PrivateIPAddressVersion' back to 'System.String'."
 "Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.Automation.RemoveAzureRmContainerService","Remove-AzureRmContainerService","0","1020","The cmdlet 'Remove-AzureRmContainerService' no longer has output type 'System.Void'.","Make cmdlet 'Remove-AzureRmContainerService' return type 'System.Void'."
+"Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.Automation.InvokeAzureRmVMRunCommand","Invoke-AzureRmVMRunCommand","0","1050","The parameter set 'VMParameter' for cmdlet 'Invoke-AzureRmVMRunCommand' has been removed.","Add parameter set 'VMParameter' back to cmdlet 'Invoke-AzureRmVMRunCommand'."
+"Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.RemoveAzureRmVmssDiagnosticsExtension","Remove-AzureRmVmssDiagnosticsExtension","0","1020","The cmdlet 'Remove-AzureRmVmssDiagnosticsExtension' no longer has output type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSet'.","Make cmdlet 'Remove-AzureRmVmssDiagnosticsExtension' return type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSet'."
+"Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.RemoveAzureRmVmssDiagnosticsExtension","Remove-AzureRmVmssDiagnosticsExtension","0","3010","The property 'IdentityIds' of type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSetIdentity' has been removed.","Add the property 'IdentityIds' back to type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSetIdentity'."
+"Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.NewAzureSshPublicKeyCommand","Add-AzureRmVMSshPublicKey","0","3000","The type of property 'Identity' of type 'Microsoft.Azure.Commands.Compute.Models.PSVirtualMachine' has changed from 'Microsoft.Azure.Management.Compute.Models.VirtualMachineIdentity' to 'Microsoft.Azure.Commands.Compute.Models.PSVirtualMachineIdentity'.","Change the type of property 'Identity' back to 'Microsoft.Azure.Management.Compute.Models.VirtualMachineIdentity'."
```

Comment:
All these lines below that say,
```
The type of property 'Identity' of type 'Microsoft.Azure.Commands.Compute.Models.PSVirtualMachine' has changed from 'Microsoft.Azure.Management.Compute.Models.VirtualMachineIdentity' to 'Microsoft.Azure.Commands.Compute.Models.PSVirtualMachineIdentity'
```
means that your `PSVirtualMachineIdentity` does not match the public interface of `VirtualMachineIdentity`. Otherwise, these exceptions should not be needed.
@cormacpayne Can you double check this? I just want to be 100% certain that the properties are checked recursively.

Reply:
There was a breaking change in Microsoft.Azure.Management.Compute.Models.VirtualMachineIdentity, in which identityIds are removed and userAssignedIdentities are added. So I added it in PSVirtualMachineIdentity.  https://github.com/hyonholee/azure-powershell/blob/ignite/src/ResourceManager/Compute/Commands.Compute/Models/PSVirtualMachineIdentity.cs#L42

Reply:
Same problem as I described above. Do not derive from your client library types. Use composition.